recipes/mold/all/conanfile.py

```diff
@@ -64,26 +70,57 @@ def _patch_sources(self):
         files.replace_in_file(self, "source_subfolder/Makefile", "MOLD_LDFLAGS += -lmimalloc", "MOLD_LDFLAGS += -L{} -lmimalloc".format(
             self.deps_cpp_info["mimalloc"].lib_paths[0]))
 
+    @functools.lru_cache(1)
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["MOLD_USE_SYSTEM_MIMALLOC"] = True
+        cmake.definitions["MOLD_USE_SYSTEM_TBB"] = True
+        cmake.definitions["MOLD_USE_MOLD"] = False
+        cmake.configure()
+        return cmake
+
+    def build_requirements(self):
+        self.tool_requires("cmake/3.23.0")
+
     def requirements(self):
         self.requires("zlib/1.2.12")
         self.requires("openssl/1.1.1q")
-        self.requires("xxhash/0.8.1")
         self.requires("onetbb/2021.3.0")
         self.requires("mimalloc/2.0.6")
+        self.requires("xxhash/0.8.1")
 
     def source(self):
         files.get(self, **self.conan_data["sources"][self.version],
                   destination=self._source_subfolder, strip_root=True)
 
     def build(self):
-        self._patch_sources()
-        with files.chdir(self, self._source_subfolder):
-            autotools = AutoToolsBuildEnvironment(self)
-            autotools.make(target="mold", args=['SYSTEM_TBB=1', 'SYSTEM_MIMALLOC=1'])
+        if Version(self.version) < "1.4":
+            self._patch_sources()
+            with files.chdir(self, self._source_subfolder):
+                autotools = AutoToolsBuildEnvironment(self)
+                autotools.make(target="mold", args=['SYSTEM_TBB=1', 'SYSTEM_MIMALLOC=1'])
+        else:
+            # Error out if ZLIB is not found, we want to be predictable
+            files.replace_in_file(self, "source_subfolder/CMakeLists.txt", "find_package(ZLIB QUIET)", "find_package(ZLIB REQUIRED)")
```

Comment:
I'll move this to a proper helper function once we agree on the topics mentioned on the PR description